Automattic/jetpack
# Contact Form: refactor form to use forms package

## Proposed changes:

As part of the Contact Form module deprecation, this PR refactors the `Grunion_Contact_Form` class to reference the Forms package.

The refactored code uses object composition to call code from the Forms package. The refactored class holds an instance of `Contact_Form` and its methods call the composed object methods. The public properties are made private, and a setter and getter are defined to get and set the properties on the composed object instead. Static methods call their counterpart in `Contact_Form`.

### Other information:


## Jetpack product discussion


Project thread: pf5801-Aj-p2
Issue: https://github.com/Automattic/jetpack/issues/35653

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Let's do some regression testing.

- Spin up a test site with this branch
- Update the file `/projects/plugins/jetpack/modules/contact-form.php` so the condition on line 33 is falsy (or comment the block). This ensures we're loading the Contact Form module.
- Create a new post
- Check you can still add a Contact Form block and that nothing breaks
- Open the preview and verify you can see and use the form
